homeassistant/core.py

```diff
@@ -918,6 +918,24 @@ def async_entity_ids(
             if state.domain in domain_filter
         ]
 
+    @callback
+    def async_entity_ids_count(
+        self, domain_filter: Optional[Union[str, Iterable]] = None
+    ) -> int:
+        """Count the entity ids that are being tracked.
+
+        This method must be run in the event loop.
+        """
+        if domain_filter is None:
+            return len(self._states.keys())
+
+        if isinstance(domain_filter, str):
+            domain_filter = (domain_filter.lower(),)
+
+        return len(
+            [state for state in self._states.values() if state.domain in domain_filter]
+        )
```

Comment:
You probably know the performance patterns better than me so this is just a question ... could it be faster to use a different type, like integers or `None`?

```suggestion
        return len(
            [None for state in self._states.values() if state.domain in domain_filter]
        )
```

Reply:
```
(venv) root@ha-dev:~/home-assistant# python -m timeit '[n for n in range(1000)]'
5000 loops, best of 5: 44.2 usec per loop
(venv) root@ha-dev:~/home-assistant# python -m timeit '[None for n in range(1000)]'
5000 loops, best of 5: 43.6 usec per loop
```

`None` does appear to be a just a bit faster 👍